### ISSUE TITLE
add -stdlib=libstdc++ to libcpptraj build osx/clang/miniconda

### ISSUE THREAD

**Issue by hainm** [Amber-MD/cpptraj#327]
hi,

So I am stuck in how to pass `-stdlib=libstdc++` that to libcpptraj build. I've tried to add the flag to $AMBERHOME/config.h
and can make pytraj/cpptraj works with clang/miniconda. But messing up with some other programs.

Any hint for introducing those flags to only libcpptraj.

Pseudocode
```python
if is_osx and hase_miniconda_python and use_clang:
    insert_that_flag_to_libcpptraj_build
```

We are thinking about encouraging AMBER users to use official `clang` and try to drop supporting macport. 

Let me know if I am not clear.

**Comment by hainm** [comment]
Please note that it's very easy to update stand alone cpptraj version: just change "CXX=clang"
to "CXX=clang -stdlib=libstdc++" in its config.h

**Comment by hainm** [comment]
need to have cpptraj's Makefile_at accepts `AMBERLDFLAGS` too

```
#  $(LIBDIR)/libcpptraj$(SHARED_SUFFIX): $(OBJECTS) pub_fft.o $(EXTERNAL_LIBS)
#        $(CXX) $(MAKE_SHARED) $(WARNFLAGS) $(LDFLAGS) -o $@ $(OBJECTS) pub_fft.o \
# -      -L$(LIBDIR) $(NETCDFLIB) $(ZLIB) $(BZLIB) $(FLIBS_PTRAJ) $(READLINE)
# +      $(AMBERLDFLAGS) -L$(LIBDIR) $(NETCDFLIB) $(ZLIB) $(BZLIB) $(FLIBS_PTRAJ) $(READLINE)
```

So pytraj can force `libcpptraj` to link to `/usr/lib/libstdc++.6.dylib` by
`export AMBERLDFLAGS=-L/usr/lib/`

**Comment by hainm** [comment]
The reason I don't want to
`export CC=clang -stdlib=libstdc++` when building whole AmberTools
because this does not play nicely with emil library.

**Comment by drroe** [comment]
@hainm Is this solely an issue within AmberTools? As far as I know the `-stdlib=libstdc++` flag should be properly set by the standalone configure.

**Comment by hainm** [comment]
> Is this solely an issue within AmberTools?

yes. Because cpptraj uses variables from $AMBERHOME/config.h and cpptraj links to other packages in amber too. 

again, pytraj has no problem with standalone cpptraj.

**Comment by hainm** [comment]
uhm I think I already fix this by set environment

https://github.com/Amber-MD/osx_install_guide/blob/master/clang.md

**Comment by drroe** [comment]
Can you see if #340 helps this at all?

**Comment by hainm** [comment]
thanks I will try.

**Comment by hainm** [comment]
still got error. I reopen #220

**Comment by drroe** [comment]
@hainm is this addressed by #379?

**Comment by hainm** [comment]
yeah. we're done (hopefully).
